evalml/objectives/standard_metrics.py

```diff
@@ -813,6 +816,47 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class MASE(TimeSeriesRegressionObjective):
+    """Mean absolute percentage error for time series regression. Scaled by 100 to return a percentage.
+
+    Only valid for nonzero inputs. Otherwise, will throw a ValueError.
+
+    Example:
+        >>> y_true = pd.Series([1.5, 2, 3, 1, 0.5, 1, 2.5, 2.5, 1, 0.5, 2])
+        >>> y_pred = pd.Series([1.5, 2.5, 2, 1, 0.5, 1, 3, 2.25, 0.75, 0.25, 1.75])
+        >>> np.testing.assert_almost_equal(MAPE().objective_function(y_true, y_pred), 15.9848484)
+    """
+
+    name = "Mean Absolute Scaled Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+    is_bounded_like_percentage = False  # Range [0, Inf)
+    expected_range = [0, float("inf")]
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        """Objective function for mean absolute percentage error for time series regression."""
+        y_train = y_true
```

Comment:
MASE takes in training data as a parameter, since we don't use training data we just used the actual data in place of training data for now, but I'm not sure if that's what we should be doing

Reply:
This brings up a very interesting question. A quick test shows that the results do change based on what you pass in to the `y_train` argument, meaning that the results we pass in with this are going to be less correct.

Looking through where we score pipelines, I don't think it would be too much extra lift to add the ability to pass `y_train` through. We have access to it in the upper level time series pipeline scoring functions, as it's pretty common for time series related things in general to require the historical data.

I'm curious what others think though - should we update our objective functions to have access to `y_train`, or should we keep it as is like this?

Reply:
I think we should update it to support MASE properly but lets work on it in a followup issue. @remyogasawara can you file the follow up in EvalML? Just basically stating the requirements that @eccabay wrote above?